source/Coop/Mod/Managers/ClientCharacterCreatorManager.cs

```diff
@@ -78,16 +82,22 @@ public override void OnLoadFinished()
         private void SkipCharacterCreation()
         {
             CharacterCreationState characterCreationState = GameStateManager.Current.ActiveState as CharacterCreationState;
-            bool flag = CharacterObject.PlayerCharacter.Culture == null;
+            bool flag = characterCreationState.CurrentStage is CharacterCreationCultureStage;
             if (flag)

```

Comment:
```suggestion
            if (characterCreationState.CurrentStage is CharacterCreationCultureStage)
```
and then remove the previous bool. No need to store this for later since it isn't used.

Reply:
And for the same checks below.